Board.cpp

```diff
@@ -190,28 +191,27 @@ void Board::step() {
 	bool isColiding = tryMoveBlock(0, 1);
 
 	if (isColiding) {
-		
+
 		if (checkIfLost()) {
 			return;
 		}
 		layBlockInField();
 		int bottom = block->getBlockMinY() - boardOffset.getY();
 		int top = block->getBlockMaxY() - boardOffset.getY();
+		auto bomb = dynamic_cast<Bomb*>(block);
+
 		if (bomb) {
-			bomb->explode(&field, boardOffset.getX(), boardOffset.getY(), HEIGHT, WIDTH);
+			bomb->explode(&field, boardOffset.getX(), boardOffset.getY(), HEIGHT-1, WIDTH-1);

```

Comment:
HEIGT AND WIDTH are counting from 1

the method uses iniices counting from 0 =)